Mathlib/Data/Fin/Basic.lean

```diff
@@ -1377,488 +1567,1212 @@ section SuccAbove
 
 /-- `succAbove p i` embeds `Fin n` into `Fin (n + 1)` with a hole around `p`. -/
 def succAbove (p : Fin (n + 1)) (i : Fin n) : Fin (n + 1) :=
-  if i.1 < p.1 then castSucc i else succ i
+  if castSucc i < p then i.castSucc else i.succ
 
-theorem strictMono_succAbove (p : Fin (n + 1)) : StrictMono (succAbove p) :=
-  (castSuccEmb : Fin n ↪o _).strictMono.ite (succEmbedding n).strictMono
-    (fun _ _ hij hj => lt_trans ((castSuccEmb : Fin n ↪o _).lt_iff_lt.2 hij) hj) fun i =>
-    (castSucc_lt_succ i).le
-#align fin.succ_above_aux Fin.strictMono_succAbove
-
-/--  `Fin.auccAbove` as an `OrderEmbedding`, `succAboveEmb p i` embeds `Fin n` into `Fin (n + 1)`
-with a hole around `p`. -/
-@[simps! apply toEmbedding]
-def succAboveEmb (p : Fin (n + 1)) : Fin n ↪o Fin (n + 1) :=
-  OrderEmbedding.ofStrictMono (succAbove p) (strictMono_succAbove p)
-#align fin.succ_above Fin.succAboveEmb
+theorem succAbove_eq_castSucc_or_succ (p : Fin (n + 1)) (i : Fin n) :
+    p.succAbove i = i.castSucc ∨ p.succAbove i = i.succ := ite_eq_or_eq _ _ _
 
 /-- Embedding `i : Fin n` into `Fin (n + 1)` with a hole around `p : Fin (n + 1)`
 embeds `i` by `castSucc` when the resulting `i.castSucc < p`. -/
-theorem succAbove_below (p : Fin (n + 1)) (i : Fin n) (h : castSucc i < p) :
-    p.succAbove i = castSucc i := if_pos h
-#align fin.succ_above_below Fin.succAbove_below
+theorem succAbove_of_castSucc_lt {p : Fin (n + 1)} {i : Fin n} (h : castSucc i < p) :
+    p.succAbove i = castSucc i := dif_pos h
+#align fin.succ_above_below Fin.succAbove_of_castSucc_lt
+theorem succAbove_of_succ_le {p : Fin (n + 1)} {i : Fin n} (h : succ i ≤ p) :
```

Comment:
blank line before theorem?

Reply:
See, Yael was telling me recently that they like to group similar theorems together.

Reply:
Personally, when grouping things I like to it with a header comment like:
```lean
/-!
## Foo theorems
-/
```